book/src/getting-started/1-qobjects-in-rust.md

```diff
@@ -0,0 +1,77 @@
+<!--
+SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+
+SPDX-License-Identifier: MIT OR Apache-2.0
+-->
+
+# QObjects in Rust
+
+> The right tool for the right job.
+
+> If you only have a hammer, every problem looks like a nail.
+
+> Don't bring a knife to a gun fight.
+
+There are many bits of advice like that.
+With CXX-Qt, we aim to make it possible to use the right tool for each of the many jobs necessary to build a modern GUI application.
+
+So what is in our toolbox for a typical Qt application?
+- QML - A declarative, flexible, dynamically-typed, interpreted language that is purpose built to define reactive and beautiful GUI layouts and widgets with quick iteration speed.
+- C++ - The traditional back-end of Qt - A fast, low-level language with a strong type system. C++ offers a rich ecosystem, many Qt-specific libraries and bare-metal performance. The cost when using C++ is that it is slow to develop, very error-prone and can easily lead to memory-issues, which can instantly crash your application and cause security issues.
+
+Notably absent then is a back-end language that allows us to get rid of the issues C++ has and provides us with a safe way to write fast Back-end code.
+This of course is where Rust comes in.
+Whilst Rust doesn't have quite as rich of an ecosystem, it is typically faster to develop than C++, with easy dependency management, and most importantly, safe memory access.
+Therefore it is an ideal candidate to replace C++ for writing the back-end business-logic code that feeds the GUI with data.
+
+However, both C++ as well as QML still have their place in Qt applications.
```

Comment:
not sure having "both" and "as well as" for the same pair works ? It needs to be "both C++ and QML ... " or "C++ as well as QML ... "

Reply:
I'll remove the both :+1: